### DISCUSSION TITLE
How to make C++ indentation correct?

### DISCUSSION THREAD

**Discussion by ck961018** [zed-industries/zed#21161] [Feature Requests]
I use zed to write C++, and when I type the following code:
```
if (true)
int a = 1;
int b = 1;
```
The indentation on the second line is incorrect.

And when I manually add indentation on the second line, the indentation on the third line will be incorrect:
```
if (true)
    int a = 1;
    int b = 1;
```

I think this should be due to the tree-sitter, and caused by the [indents.scm](https://github.com/zed-industries/zed/blob/main/crates/languages/src/cpp/indents.scm) being too simple.

**Comment by wojdyr** [comment]
I'm also struggling with incorrect indents in C/C++ and I guess you are right, the indents.scm file for nvim is much longer:
https://github.com/nvim-treesitter/nvim-treesitter/blob/master/queries/c/indents.scm
I'll open a feature request.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
